### PR TITLE
Convolutional Neural ODE for MNIST Classification Demo

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: docs/src/examples/mnist_conv_neural_ode.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+### Train
+
+To train our model, we select the appropriate trainable parameters of our network with `params`.
+In our case, backpropagation is required for `down`, `nn_ode` and `fc`. Notice that the parameters
+for Neural ODE is given by `nn_ode.p`:
+
+```julia
+# Train the NN-ODE and monitor the loss and weights.
+Flux.train!(loss, params(down, nn_ode.p, fc), train_dataloader, opt, cb = cb)
+```
```
Comment (ChrisRackauckas) [inline docs/src/examples/mnist_conv_neural_ode.md:368]: show example training print outs?

**Comment by ChrisRackauckas** [comment]
Add a GPU on this? I want to have something that ensures future updates don't cause OOMs.

**Comment by avik-pal** [comment]
Will add the test. We might want to wait for the CUDA3.0 compatible Flux release before merging.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
@jessebett does this look good? Do you have a PyTorch version of this to benchmark against?

**Comment by ChrisRackauckas** [comment]
Beautiful. Okay, let's wait for the update.

**Comment by DhairyaLGandhi** [comment]
@avik-pal could you post the runtime numbers with cuda 3 as well? Thanks!

**Comment by avik-pal** [comment]
1 epoch with 512 batch size takes 22s with CUDA3.0 while the released version of the Flux takes around 80-90s. The numbers are with logging removed

File: docs/src/examples/mnist_conv_neural_ode.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+`nn`: A 2 layer Convolutional Neural Network Chain with `tanh` activation which is used to model
+      our differential equation
+
+`nn_ode`: ODE solver layer
+
+`fc`: The final fully connected layer which maps our learned features to the probability of
+      the feature vector of belonging to a particular class
+
+`|> gpu`: An utility function which transfers our model to GPU, if it is available
```
Comment (DhairyaLGandhi) [inline docs/src/examples/mnist_conv_neural_ode.md:229]:
```suggestion
`gpu`: An utility function which transfers our model to GPU, if one is available
```

**Comment by avik-pal** [comment]
@ChrisRackauckas flux with cuda3.0 support is released. Can you rerun buildkite? (I don't have the permissions)
